sway-lib-std/src/vec.sw

```diff
@@ -142,3 +145,84 @@ impl<T> Vec<T> {
         self.len == 0
     }
 }
+
+/// TODO: Add proper errors
+
+pub struct StorageVec<V> {}
```

Comment:
I will argue that this needs to be inside `storage.sw` instead. 
Thoughts @otrho @adlerjohn @nfurfaro ?

Reply:
Yeah probably. The `storage` library is growing though, it may be time to start splitting it into sub-libraries and calling it something other than `storage` because `storage` is a keyword 😄

Reply:
I would be inclined to put it in `storage.sw` instead of `vec.sw` given those two choices. A separate file might better in the long-term, but we can worry about that later.

Reply:
Agreed 👍

Reply:
late to the party, but agreed as well.